src/lib/app-state-hoc.jsx

```diff
@@ -37,8 +40,21 @@ const AppStateHOC = function (WrappedComponent) {
             } else {
                 intl = intlInitialState;
             }
+            if (props.mode) {
+                mode = {
+                    mode: props.mode
+                };
+            } else {
+                mode = modeInitialState;
+            }
 
-            this.store = createStore(reducer, intl, enhancer);
+            this.store = createStore(
+                reducer,
+                {
+                    intl: intl,
+                    guiMode: mode
```

Comment:
Curious about `mode` vs `guiMode`. I think I agree that `mode` might be too ambiguous. Since it's distinct from full screen mode, maybe we should stick to that and just call it `isPlayerMode` similar to `isFullScreen`?

Reply:
Yes ambiguity, and `state.mode.mode` just looks wrong.